### PR TITLE
Changed release process to use a GitHub Actions workflow

### PR DESCRIPTION
**Pull request by andy-maier**

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+          echo "Update version '${{ steps.set-update-version.outputs.result }}' in tag '${{ steps.set-tag.outputs.result }}' is invalid (must be non-0 when releasing stable branch for pushed tag)";
+          false;
+        fi
+
+    #-------- Setup of work environment
+    - name: Checkout repo
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+    - name: Set up Python 3.10
```
[the remark this answers is not in the document]
Comment (andy-maier) [reply .github/workflows/publish.yml:126]: It was the latest at the time I created the first version of this workflow :-)
It really does not matter much, except that it should not be sufficiently old to be problematic.
I have changed it to 3.12.

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+        make develop
+
+    #-------- Publishing of package
+    - name: Build the distribution
+      run: |
+        make build
+    - name: Display the distribution directory
+      run: |
+        ls -l dist
+    # - name: Publish distribution to TestPyPI
```
Comment (KSchopmeyer) [inline .github/workflows/publish.yml:141]: This group of comments referencing TestPyPi confuses me.

**Comment by KSchopmeyer** [comment]
I have reviewed this and made one more comment.  Except for the issue of the new errors (check or pylint again) I am ready to pass it with resolution of that comment.

File: .github/workflows/publish.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (KSchopmeyer) [reply .github/workflows/publish.yml]: Agreed..

File: .github/workflows/publish.yml (hunk 2 above)
Comment (andy-maier) [reply .github/workflows/publish.yml:141]: It is there so it can be enabled for testing or debugging the release process.

**Pull request merged**